### PR TITLE
feat: cdw-40 endpoint to store and retrieve users

### PR DESCRIPTION
**Pull request by ifasama**

### CODE REVIEW

File: backend/src/api/v1/mentorShip/services/types.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export interface userDetails {
+    firstName: string
+    lastName: string
+    email: string
+    username: string
+    password: string
+    location: string
+}
```
Comment (jtabba) [inline backend/src/api/v1/mentorShip/services/types.ts]: What were these included for?

File: backend/src/api/v1/mentorShip/services/userServices.ts [hunk 2]
```diff
@@ -2,39 +2,31 @@ import { ObjectId } from 'mongodb'
...
 
-export const getSingleUser = async (id: string) => {
-    const collection = await getCollection('users')
-    const userData = await collection?.findOne({
-        _id: new ObjectId(id)
+export const getSingleUser = async (args: userInfo) => {
+    const userList = await getCollection('users')
+    const userData = await userList?.findOne({
+        id: new ObjectId(args.id)
     })
```
Comment (jtabba) [inline backend/src/api/v1/mentorShip/services/userServices.ts:21]: We need to fetch a user by their email as it is the only unique identifier we have

File: backend/src/api/v1/mentorShip/services/userServices.ts [hunk 3]
```diff
@@ -2,39 +2,31 @@ import { ObjectId } from 'mongodb'
...
     return userData
 }
 
-export const postSingleUser = async (args: userDetail) => {
-    const collection = await getCollection('users')
-    const usernameExist = await collection?.findOne({
+export const postSingleUser = async (args: userDetails) => {
+    const userList = await getCollection('users')
+    const usernameExist = await userList?.findOne({
         username: args.username
```
Comment (jtabba) [inline backend/src/api/v1/mentorShip/services/userServices.ts]: Use email instead

File: backend/src/api/v1/mentorShip/services/userServices.ts [hunk 4]
```diff
@@ -65,3 +57,23 @@ export const postSingleUser = async (args: userDetail) => {
...
+
+export const postUserFromAuth = async (args: userInfo) => {
+    const userList = await getCollection('users')
+    const insertedData = await userList?.insertOne({
+        name: args.name,
+        email: args.email,
+        dateOfBirth: args.dateOfBirth,
+        yearsOfExperience: args.yearsOfExperience,
+        currentJobTitle: args.currentJobTitle,
+        AreasOfExpertise: args.AreasOfExpertise,
```
Comment (jtabba) [inline backend/src/api/v1/mentorShip/services/userServices.ts]: The A should be lowercase (that's my bad)

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,14 +1,15 @@
 import { useEffect, useState } from "react";
-import { getAllUsers } from "./requests/getAllUsers";
+import { getAllUsers, getSingleUser } from "./requests/getAllUsers";
 import { userData } from "./requests/types";
 import { UserInterface } from "./components/UserInterface";
 import { Routes, Route, BrowserRouter as Router } from "react-router-dom";
-import Home from "./pages/home";
-import Profile from "./pages/profile";
+import Home from "./pages/Home";
+import Profile from "./pages/Profile";
```
Comment (jtabba) [inline frontend/src/App.tsx]: Pages are lowercase as we discussed in Discord

File: frontend/src/App.tsx [hunk 6]
```diff
@@ -18,15 +19,33 @@ const App = () => {
...
-		console.log("users", users);
-	});
+		console.log("users", users); // this should be deleted later to avoid non stop render
+	}, [users]);
 	// ***PURELY FOR EXAMPLE***
 
+	useEffect(() => {
+		// open browser console to see result
+		const getUser = async () => {
+			const response = await getSingleUser();
```
Comment (jtabba) [inline frontend/src/App.tsx]: Just for reference we would need to use the users email to retrieve their data

File: frontend/src/App.tsx [hunk 7]
```diff
@@ -18,15 +19,33 @@ const App = () => {
...
+			} else {
+				throw Error("Cannot fetch"); // temporary
+				// create error notification wth Chakra
+			}
+		};
+
+		getUser();
+		console.log("users", singleUser); // this should be deleted later to avoid non stop render
+	}, [singleUser]);
+
```
Comment (jtabba) [inline frontend/src/App.tsx]: You've done well with this @ifasama but the front end implementation was not part of this ticket (Noah is currently working on it). This ticket was solely concerned with creating the endpoint. Please remove any front end changes

File: frontend/src/requests/getAllUsers.ts [hunk 8]
```diff
@@ -27,3 +27,23 @@ export const getAllUsers = async (): Promise<IResponse | IErrorResponse> => {
...
+		}
+	} catch (err) {
+		console.error(err);
+
+		return {
+			success: false,
+			data: isAxiosError(err) ? err.message: (err as string)
+		}
+	}
+}
```
Comment (jtabba) [inline frontend/src/requests/getAllUsers.ts]: As above, not part of this ticket. Please remove

**Review by jtabba — CHANGES_REQUESTED** [review]
Great work overall Ifa. I've left comments about required changes - if they are confusing let me know and we can set up a call via discord

**Comment by ifasama** [comment]
Fix merge conflict after rebase. Needed to forced push but hope it is ok.

**Review by jtabba — APPROVED** [review]

**Pull request merged**